components/centraldashboard/app/k8s_service.ts

```diff
@@ -0,0 +1,36 @@
+import * as k8s from '@kubernetes/client-node';
+
+/** Wrap Kubernetes API calls in a simpler interface for use in routes. */
+export class KubernetesService {
+  private kubeConfig: k8s.KubeConfig;
+  private k8sApi: k8s.Core_v1Api;
+
+  constructor() {
+    this.kubeConfig = new k8s.KubeConfig();
+    console.info('Initializing Kubernetes configuration');
+    this.kubeConfig.loadFromDefault();
+    this.k8sApi = this.kubeConfig.makeApiClient(k8s.Core_v1Api);
+  }
+
+  /** Retrieves the list of namespaces from the Cluster. */
+  async getNamespaces(): Promise<k8s.V1Namespace[]> {
+    try {
+      const {body} = await this.k8sApi.listNamespace();
+      return body.items;
+    } catch (err) {
+      console.error('Unable to fetch Namespaces', err.body);
+      return [];
```

Comment:
_**[idiomatic js]**_
Shouldn't we reject the promise here, instead of returning an empty array?

Reply:
Normally I think that would be the best approach, but if we were to do so here we'd need to add some additional handling to the client code. With this approach, we can log the errors for investigation and not affect the behavior of the front-end for the read-only data being shown.

Reply:
Fait enough, lgtm